### PR TITLE
Integrate LLVM at llvm/llvm-project@bd92e4620433

### PR DESCRIPTION
**Pull request by abhigunj**

### CODE REVIEW

File: stablehlo/transforms/StablehloAggressiveFolder.cpp [hunk 1]
```diff
@@ -79,8 +79,9 @@ APSInt getAPSInt(Type type, uint64_t value) {
   } else {
     llvm::report_fatal_error("expected integer type");
   }
-  return APSInt({/*numBits=*/numBits, value},
-                /*isUnsigned=*/isUnsigned);
+  return APSInt(
+      {/*numBits=*/numBits, value, /*isSigned=*/false, /*implicitTrunc=*/true},
```
Comment (ghpvnist) [inline stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: @sdasgup3 Should this be `/*isSigned=integerType.isSignedInteger()*/`?

Comment (sdasgup3) [reply stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: That is right. Also for StableHLO, `signless`  is also treated as signed, so how about we use `/*isSigned=*/!isUnSigned`.

**Review by sdasgup3 — APPROVED** [review]

File: stablehlo/transforms/StablehloAggressiveFolder.cpp (hunk 1 above)
Comment (ghpvnist) [reply stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: Setting that to `!isUnsigned` appears to break `EvalCompareOpPattern`: https://github.com/tensorflow/mlir-hlo/commit/5eef965834ba46c70d74255b278a9bdd3df697a0

Comment (abhigunj) [reply stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: This change is back-port to gh from temporary.patch and unrelated to LLVM bump. To unblock the integrate I am temporarily marking this comment resolved and will reopen to discuss it further.

Comment (sdasgup3) [reply stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: Based on https://github.com/tensorflow/mlir-hlo/commit/5eef965834ba46c70d74255b278a9bdd3df697a0, lets keep the status quo.

Comment (abhigunj) [reply stablehlo/transforms/StablehloAggressiveFolder.cpp:83]: keeping `/*isSigned=*/false` unchanged as `!isUnsigned` can't be used.

**Pull request merged**